datafusion-examples/examples/advanced_parquet_index.rs

```diff
@@ -504,7 +504,7 @@ impl TableProvider for IndexTableProvider {
             .with_file(partitioned_file);
 
         // Finally, put it all together into a DataSourceExec
-        Ok(file_scan_config.new_exec())
+        Ok(file_scan_config.build())
```

Comment:
This is basically the change -- use `build` rather than `new_exec` which I think is more consistent with the style of code in the rest of this crate (and the Rust builder style in general)